<launch>
    <arg name="visualization" default="true"/>
    <node pkg="tf" type="static_transform_publisher" name="world_map" args="0 0 0 0 0 0 world map 10"/>
    <node pkg="adaptive_viewutility" type="test_utility_map_node" name="test_grid" output="screen">
        <param name="terrain_mesh_path" value="$(find adaptive_viewutility)/../resources/landscape_mountains_v5.obj"/>
        <param name="output_file_path" value="$(find adaptive_viewutility)/../output/map_data.csv"/>
        <!-- <param name="min_traingulation_angle" value="1.0"/> -->
        <param name="reference_view_distance" value="150.0"/>
        <!-- <param name="sigma_k" value=/> -->
    </node>

    <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
        <rosparam command="load" file="$(find adaptive_viewutility)/launch/visualize.yaml" />
    </node>

    <group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find adaptive_viewutility)/launch/config_utility.rviz" />
    </group>
</launch>
